google/resource_compute_network.go

```diff
@@ -64,6 +69,39 @@ func resourceComputeNetwork() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"peering": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateGCPName,
+						},
+						"network": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateFunc:     validateRegexp(peerNetworkLinkRegex),
+							DiffSuppressFunc: peerNetworkLinkDiffSuppress,
```

Comment:
I have no preferences here personally. I think that we could migrate between the two at will without even using a `MigrateFunc` if we ever wanted to, too.